src/app/components/timeline/CardDialog.js

```diff
@@ -57,9 +70,62 @@ class CardDialog extends Component {
     this.props.closeDialog()
   }
 
+  addInputRef = (ref) => {
+    // Set state needs to be queued so that React does not overwrite
+    // the attribute in a merge
+    setTimeout(() => {
+      this.setState({
+        inputRefs: [...this.state.inputRefs, ref],
+      })
+    }, 0)
+  }
+
+  findChildInput = (id) => {
+    const foundInput = this.state.inputRefs.find((ref) => {
+      return ref && ref.props.id === id
+    })
+    return foundInput && findDOMNode(foundInput)
+  }
```

Comment:
This searches for a child ref which matches the name of the field we're looking for.